Workaround for sit-for bug during edts-api-start-server

henrikcaesar opened the pull request:
There is a bug in sit-for that make the call return immediately
instead of waiting the wanted numer of seconds. This bug has been
around since emacs-23.2 and was fixed Wed, 29 Jul 2015, see:

  https://debbugs.gnu.org/cgi/bugreport.cgi?bug=15990
  https://debbugs.gnu.org/cgi/bugreport.cgi?bug=20935

This bug make EDTS fail to start with the message:

  EDTS: Could not start main server

since the retry loop in edts-api-start-server is executed too fast,
before the web server component has started.

Since a many emacs version are affected many users might get into
trouble for a long period of time. The fix not even been included in a
official emacs release yet. To work around the problem instead of
relying on sit-for also check how long time have passed in the while
loop condition.

henrikcaesar commented:
Hi,

I can't seem to get the tests to work, and I am unable to run them on my system as it give me another error (added at the end).

I had a look at the tests and I can not see directly how the change I made caused the test to fail. The last build on master passed 4 months ago, so I assume the tests still should pass.

I might need to update the Pull Request with some debug printouts in order to reproduce and solve the problem. I will investigate further. If you have any comments about the change please let me know before I spend too much time on this.

Thanks and regards,
Henrik Caesar

==> edts (ct)
Loading erlang-skels...
EDTS [info]: Initializing plugin edts_debug
EDTS [info]: Initializing plugin edts_dialyzer
EDTS [info]: Initializing plugin edts_xref
Loading /home/ehencae/src/edts/elisp/edts/edts-project-test.el (source)...
Loading /home/ehencae/src/edts/elisp/edts/edts-shell-test.el (source)...
Loading /home/ehencae/src/edts/elisp/edts/edts-test.el (source)...
Running 1 tests (2015-12-16 10:58:07+0100)
   passed  1/1  edts-shell-make-comint-buffer-test

Ran 1 tests, 1 results as expected (2015-12-16 10:58:07+0100)

Wrote /home/ehencae/.emacs.d/ac-comphist.dat
Loading erlang-skels...
EDTS [info]: Initializing plugin edts_debug
EDTS [info]: Initializing plugin edts_dialyzer
EDTS [info]: Initializing plugin edts_xref
Loading /home/ehencae/src/edts/elisp/edts/edts-project-test.el (source)...
Loading /home/ehencae/src/edts/elisp/edts/edts-shell-test.el (source)...
Loading /home/ehencae/src/edts/elisp/edts/edts-test.el (source)...
EDTS [info]: Running test suite: edts-project-suite
EDTS [error]: Error running tests
  backtrace()
  (progn (edts-log-error "Error running tests") (backtrace) (kill-emacs 2))
  (unwind-protect (let ((exit-status 0)) (let ((--dolist-tail-- edts-test-suite-alist) suite) (while --dolist-tail-- (setq suite (car --dolist-tail--)) (let\* ((suite-name (car suite)) (stats (edts-test-run-suite-batch suite-name))) (if (zerop (ert-stats-completed-unexpected stats)) nil (setq exit-status 1))) (setq --dolist-tail-- (cdr --dolist-tail--)))) (kill-emacs exit-status)) (progn (edts-log-error "Error running tests") (backtrace) (kill-emacs 2)))
  edts-test-run-suites-batch-and-exit()
  command-line-1(("-L" "/home/ehencae/src/edts" "-l" "test/load-tests.el" "-f" "edts-test-run-suites-batch-and-exit"))
  command-line()
  normal-top-level()
make: **\* [integration-tests] Error 2

henrikcaesar commented:
Hi again,

I created my own travisci instance for my fork of EDTS and triggered a test on master and it fails the same way as the tests for this Pull Request. So I don't think it is my change that are causing the failures.
Do you have any thoughts about the change?

Regards,
Henrik Caesar

henrikcaesar commented:
Hi, do you have any comments about the change? Should I create an issue instead? I see this problem when I run emacs 24.5.1 and have been running the workaround now for a month without any problem.

Regards,
Henrik

tjarvstrand commented:
I believe that this is the root cause of many issues opened here so thanks for looking into fixing it. However, I'd like to keep this stuff as dirt simple as possible. I'm not able to reproduce this behaviour on my machine but two possible alternative causes come to mind:
Looking at the doc for `sit-for` it states "Stop when input is available" which I'm not entirely sure if it is of consequence here. Just to be sure, does this work for you?

```
(redisplay)
(sleep-for edts-api-server-start-retry-interval)
```

How about running the shell command directly `(shell-command "sleep ...")`?

henrikcaesar commented:
I don't understand this bug... when I created this PR I had this problem all the time, and this PR fixed it. Then a few weeks ago I tried master branch EDTS again and to my surprise it worked, and therefore I haven't been able to test your suggestion... until now when the problem is back again "#!"#"$

sleep-for seem to do the trick also so I will update the PR!

Below are logs  for the startup using sit-for and sleep-for. It is clearly visible that sit-for doesn't work.

sit-for

```
EDTS (2016-02-22 15:04:51.310075086) [info]: Initializing plugin edts_dialyzer
EDTS (2016-02-22 15:04:51.311107820) [info]: Initializing plugin edts_xref
For information about GNU Emacs and the GNU system, type C-h C-a.
EDTS (2016-02-22 15:04:51.570440066) [debug]: Setting up edts-mode in buffer sgw_gtpu_term_tests.erl
EDTS (2016-02-22 15:04:51.584326964) [debug]: Sending GET-request to http://0:4587/nodes
Using a proxy for http...
Package iswitchb is obsolete!
Reading machine info...done
EDTS (2016-02-22 15:04:51.651160906) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.651464203) [info]: Failed to start EDTS server: 150 retries left
EDTS (2016-02-22 15:04:51.651738919) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.654948261) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.655228430) [info]: Failed to start EDTS server: 149 retries left
EDTS (2016-02-22 15:04:51.655478297) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.656815927) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.657035422) [info]: Failed to start EDTS server: 148 retries left
EDTS (2016-02-22 15:04:51.657199418) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.658387767) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.658639033) [info]: Failed to start EDTS server: 147 retries left
EDTS (2016-02-22 15:04:51.658867599) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.659995873) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.660242955) [info]: Failed to start EDTS server: 146 retries left
EDTS (2016-02-22 15:04:51.660520100) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.661566109) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.661817233) [info]: Failed to start EDTS server: 145 retries left
EDTS (2016-02-22 15:04:51.662045694) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.663182474) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.663407837) [info]: Failed to start EDTS server: 144 retries left
EDTS (2016-02-22 15:04:51.663595900) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.664805808) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.665053437) [info]: Failed to start EDTS server: 143 retries left
EDTS (2016-02-22 15:04:51.665222962) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.667541339) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.668020948) [info]: Failed to start EDTS server: 142 retries left
EDTS (2016-02-22 15:04:51.668218957) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.669333325) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.669532750) [info]: Failed to start EDTS server: 141 retries left
EDTS (2016-02-22 15:04:51.669695080) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.672139438) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.672384945) [info]: Failed to start EDTS server: 140 retries left
EDTS (2016-02-22 15:04:51.672580439) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.673645054) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.673864046) [info]: Failed to start EDTS server: 139 retries left
EDTS (2016-02-22 15:04:51.674047234) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.675098395) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.675337421) [info]: Failed to start EDTS server: 138 retries left
EDTS (2016-02-22 15:04:51.675554946) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.676628573) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.676855394) [info]: Failed to start EDTS server: 137 retries left
EDTS (2016-02-22 15:04:51.677051132) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.678114137) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.678338620) [info]: Failed to start EDTS server: 136 retries left
EDTS (2016-02-22 15:04:51.678519985) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.679571304) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.679810311) [info]: Failed to start EDTS server: 135 retries left
EDTS (2016-02-22 15:04:51.680032143) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.681078674) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.681290340) [info]: Failed to start EDTS server: 134 retries left
EDTS (2016-02-22 15:04:51.681442712) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.683006923) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.683399307) [info]: Failed to start EDTS server: 133 retries left
EDTS (2016-02-22 15:04:51.683725388) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.685434091) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.685643052) [info]: Failed to start EDTS server: 132 retries left
EDTS (2016-02-22 15:04:51.685787174) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.688510101) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.688747165) [info]: Failed to start EDTS server: 131 retries left
EDTS (2016-02-22 15:04:51.688904007) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.718466271) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.718700713) [info]: Failed to start EDTS server: 130 retries left
EDTS (2016-02-22 15:04:51.718857738) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.721437361) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.721945737) [info]: Failed to start EDTS server: 129 retries left
EDTS (2016-02-22 15:04:51.722407494) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.724096674) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.724307967) [info]: Failed to start EDTS server: 128 retries left
EDTS (2016-02-22 15:04:51.724485204) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.725608196) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.725852415) [info]: Failed to start EDTS server: 127 retries left
EDTS (2016-02-22 15:04:51.726080494) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.727199990) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.727407672) [info]: Failed to start EDTS server: 126 retries left
EDTS (2016-02-22 15:04:51.727565322) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.728707978) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.728900444) [info]: Failed to start EDTS server: 125 retries left
EDTS (2016-02-22 15:04:51.729051970) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.730001091) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.730325378) [info]: Failed to start EDTS server: 124 retries left
EDTS (2016-02-22 15:04:51.730542839) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.731626892) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.731839058) [info]: Failed to start EDTS server: 123 retries left
EDTS (2016-02-22 15:04:51.731990423) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.733067978) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.733263571) [info]: Failed to start EDTS server: 122 retries left
EDTS (2016-02-22 15:04:51.733415428) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.734491453) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.734741908) [info]: Failed to start EDTS server: 121 retries left
EDTS (2016-02-22 15:04:51.734916065) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.735969518) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.736182273) [info]: Failed to start EDTS server: 120 retries left
EDTS (2016-02-22 15:04:51.736334824) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.739489989) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.739994918) [info]: Failed to start EDTS server: 119 retries left
EDTS (2016-02-22 15:04:51.740458520) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.742120473) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.742317118) [info]: Failed to start EDTS server: 118 retries left
EDTS (2016-02-22 15:04:51.742469875) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.743418854) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.743625167) [info]: Failed to start EDTS server: 117 retries left
EDTS (2016-02-22 15:04:51.743777364) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.744899919) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.745101667) [info]: Failed to start EDTS server: 116 retries left
EDTS (2016-02-22 15:04:51.745257200) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.746497596) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.746738646) [info]: Failed to start EDTS server: 115 retries left
EDTS (2016-02-22 15:04:51.746973691) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.748115081) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.748353651) [info]: Failed to start EDTS server: 114 retries left
EDTS (2016-02-22 15:04:51.748571438) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.749665882) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.749904694) [info]: Failed to start EDTS server: 113 retries left
EDTS (2016-02-22 15:04:51.750137657) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.751088895) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.751278239) [info]: Failed to start EDTS server: 112 retries left
EDTS (2016-02-22 15:04:51.751472517) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.752336295) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.752520821) [info]: Failed to start EDTS server: 111 retries left
EDTS (2016-02-22 15:04:51.752705983) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.753655190) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.753936697) [info]: Failed to start EDTS server: 110 retries left
EDTS (2016-02-22 15:04:51.756100062) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.757308965) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.757509540) [info]: Failed to start EDTS server: 109 retries left
EDTS (2016-02-22 15:04:51.757698687) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.758530333) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.758712706) [info]: Failed to start EDTS server: 108 retries left
EDTS (2016-02-22 15:04:51.758900652) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.759736817) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.759921384) [info]: Failed to start EDTS server: 107 retries left
EDTS (2016-02-22 15:04:51.760112001) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.760938360) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.761124042) [info]: Failed to start EDTS server: 106 retries left
EDTS (2016-02-22 15:04:51.761312290) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.762164428) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.762355529) [info]: Failed to start EDTS server: 105 retries left
EDTS (2016-02-22 15:04:51.762554621) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.763389155) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.763572334) [info]: Failed to start EDTS server: 104 retries left
EDTS (2016-02-22 15:04:51.763771328) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.764610556) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.764794517) [info]: Failed to start EDTS server: 103 retries left
EDTS (2016-02-22 15:04:51.764984066) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.765813651) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.765996743) [info]: Failed to start EDTS server: 102 retries left
EDTS (2016-02-22 15:04:51.766185639) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.766999825) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.767182878) [info]: Failed to start EDTS server: 101 retries left
EDTS (2016-02-22 15:04:51.767371795) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.768221039) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.768410970) [info]: Failed to start EDTS server: 100 retries left
EDTS (2016-02-22 15:04:51.768599415) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.769447119) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.769646973) [info]: Failed to start EDTS server: 99 retries left
EDTS (2016-02-22 15:04:51.769799060) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.771101142) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.771427692) [info]: Failed to start EDTS server: 98 retries left
EDTS (2016-02-22 15:04:51.773380304) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.774462707) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.774665654) [info]: Failed to start EDTS server: 97 retries left
EDTS (2016-02-22 15:04:51.774857419) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.775727714) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.775923188) [info]: Failed to start EDTS server: 96 retries left
EDTS (2016-02-22 15:04:51.776112018) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.776960013) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.777151582) [info]: Failed to start EDTS server: 95 retries left
EDTS (2016-02-22 15:04:51.777342720) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.778188052) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.778372484) [info]: Failed to start EDTS server: 94 retries left
EDTS (2016-02-22 15:04:51.778561106) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.779390020) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.779573849) [info]: Failed to start EDTS server: 93 retries left
EDTS (2016-02-22 15:04:51.779772215) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.780600906) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.780785456) [info]: Failed to start EDTS server: 92 retries left
EDTS (2016-02-22 15:04:51.780974320) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.781825407) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.782008698) [info]: Failed to start EDTS server: 91 retries left
EDTS (2016-02-22 15:04:51.782197802) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.783029497) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.783214644) [info]: Failed to start EDTS server: 90 retries left
EDTS (2016-02-22 15:04:51.783404015) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.784267534) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.784455067) [info]: Failed to start EDTS server: 89 retries left
EDTS (2016-02-22 15:04:51.784646777) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.785481090) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.785664914) [info]: Failed to start EDTS server: 88 retries left
EDTS (2016-02-22 15:04:51.785849128) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.786782394) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.787003361) [info]: Failed to start EDTS server: 87 retries left
EDTS (2016-02-22 15:04:51.787189004) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.789021938) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.789314736) [info]: Failed to start EDTS server: 86 retries left
EDTS (2016-02-22 15:04:51.789530239) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.791757705) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.791967746) [info]: Failed to start EDTS server: 85 retries left
EDTS (2016-02-22 15:04:51.792164105) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.793085142) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.793276172) [info]: Failed to start EDTS server: 84 retries left
EDTS (2016-02-22 15:04:51.793466640) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.794304171) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.794489181) [info]: Failed to start EDTS server: 83 retries left
EDTS (2016-02-22 15:04:51.794679791) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.795512147) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.795705338) [info]: Failed to start EDTS server: 82 retries left
EDTS (2016-02-22 15:04:51.795896565) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.796725899) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.796909448) [info]: Failed to start EDTS server: 81 retries left
EDTS (2016-02-22 15:04:51.797098760) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.797999114) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.798197412) [info]: Failed to start EDTS server: 80 retries left
EDTS (2016-02-22 15:04:51.798385469) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.799249010) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.799435143) [info]: Failed to start EDTS server: 79 retries left
EDTS (2016-02-22 15:04:51.799638710) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.800489372) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.800675135) [info]: Failed to start EDTS server: 78 retries left
EDTS (2016-02-22 15:04:51.800858657) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.801703598) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.801888342) [info]: Failed to start EDTS server: 77 retries left
EDTS (2016-02-22 15:04:51.802081282) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.802910910) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.803098408) [info]: Failed to start EDTS server: 76 retries left
EDTS (2016-02-22 15:04:51.803248963) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.804376909) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.804677756) [info]: Failed to start EDTS server: 75 retries left
EDTS (2016-02-22 15:04:51.804944683) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.806841895) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.807040588) [info]: Failed to start EDTS server: 74 retries left
EDTS (2016-02-22 15:04:51.808282947) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.809321157) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.809520486) [info]: Failed to start EDTS server: 73 retries left
EDTS (2016-02-22 15:04:51.809715218) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.810581146) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.810766975) [info]: Failed to start EDTS server: 72 retries left
EDTS (2016-02-22 15:04:51.810956624) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.811810445) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.812000534) [info]: Failed to start EDTS server: 71 retries left
EDTS (2016-02-22 15:04:51.812189590) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.813023530) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.813214514) [info]: Failed to start EDTS server: 70 retries left
EDTS (2016-02-22 15:04:51.813404585) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.814252576) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.814445562) [info]: Failed to start EDTS server: 69 retries left
EDTS (2016-02-22 15:04:51.814636734) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.815479292) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.815685126) [info]: Failed to start EDTS server: 68 retries left
EDTS (2016-02-22 15:04:51.815878264) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.816721050) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.816911716) [info]: Failed to start EDTS server: 67 retries left
EDTS (2016-02-22 15:04:51.817101413) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.817942891) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.818128741) [info]: Failed to start EDTS server: 66 retries left
EDTS (2016-02-22 15:04:51.818320769) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.819151362) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.819343047) [info]: Failed to start EDTS server: 65 retries left
EDTS (2016-02-22 15:04:51.819493700) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.820445328) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.820652496) [info]: Failed to start EDTS server: 64 retries left
EDTS (2016-02-22 15:04:51.820845985) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.822582026) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.822870871) [info]: Failed to start EDTS server: 63 retries left
EDTS (2016-02-22 15:04:51.823085178) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.825275269) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.825494387) [info]: Failed to start EDTS server: 62 retries left
EDTS (2016-02-22 15:04:51.825695342) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.826635076) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.826833008) [info]: Failed to start EDTS server: 61 retries left
EDTS (2016-02-22 15:04:51.827023336) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.827885077) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.828079978) [info]: Failed to start EDTS server: 60 retries left
EDTS (2016-02-22 15:04:51.828272953) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.829115502) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.829301119) [info]: Failed to start EDTS server: 59 retries left
EDTS (2016-02-22 15:04:51.829492678) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.830333795) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.830547038) [info]: Failed to start EDTS server: 58 retries left
EDTS (2016-02-22 15:04:51.830795268) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.831939170) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.832218570) [info]: Failed to start EDTS server: 57 retries left
EDTS (2016-02-22 15:04:51.832390069) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.833632460) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.833858372) [info]: Failed to start EDTS server: 56 retries left
EDTS (2016-02-22 15:04:51.834027351) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.835168507) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.835427007) [info]: Failed to start EDTS server: 55 retries left
EDTS (2016-02-22 15:04:51.835646059) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.836865226) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.837157027) [info]: Failed to start EDTS server: 54 retries left
EDTS (2016-02-22 15:04:51.837387054) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.839311714) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.839585900) [info]: Failed to start EDTS server: 53 retries left
EDTS (2016-02-22 15:04:51.839854486) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.841039999) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.841257526) [info]: Failed to start EDTS server: 52 retries left
EDTS (2016-02-22 15:04:51.841441725) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.843764938) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.844030026) [info]: Failed to start EDTS server: 51 retries left
EDTS (2016-02-22 15:04:51.844216422) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.845607382) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.845879791) [info]: Failed to start EDTS server: 50 retries left
EDTS (2016-02-22 15:04:51.846088005) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.847429854) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.847710502) [info]: Failed to start EDTS server: 49 retries left
EDTS (2016-02-22 15:04:51.847972295) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.849036436) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.849233261) [info]: Failed to start EDTS server: 48 retries left
EDTS (2016-02-22 15:04:51.849386912) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.850636830) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.850924153) [info]: Failed to start EDTS server: 47 retries left
EDTS (2016-02-22 15:04:51.851192910) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.852363085) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.852600813) [info]: Failed to start EDTS server: 46 retries left
EDTS (2016-02-22 15:04:51.852771581) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.854824199) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.855447123) [info]: Failed to start EDTS server: 45 retries left
EDTS (2016-02-22 15:04:51.855938321) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.857918486) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.858272821) [info]: Failed to start EDTS server: 44 retries left
EDTS (2016-02-22 15:04:51.860209520) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.861728145) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.862047853) [info]: Failed to start EDTS server: 43 retries left
EDTS (2016-02-22 15:04:51.862361781) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.863672676) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.863981754) [info]: Failed to start EDTS server: 42 retries left
EDTS (2016-02-22 15:04:51.864289076) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.865596964) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.865904612) [info]: Failed to start EDTS server: 41 retries left
EDTS (2016-02-22 15:04:51.866190863) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.867748815) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.868092872) [info]: Failed to start EDTS server: 40 retries left
EDTS (2016-02-22 15:04:51.868423124) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.869950323) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.870319118) [info]: Failed to start EDTS server: 39 retries left
EDTS (2016-02-22 15:04:51.870638214) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.873749765) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.874164603) [info]: Failed to start EDTS server: 38 retries left
EDTS (2016-02-22 15:04:51.874503540) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.876171264) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.876523102) [info]: Failed to start EDTS server: 37 retries left
EDTS (2016-02-22 15:04:51.878471194) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.880922333) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.881274088) [info]: Failed to start EDTS server: 36 retries left
EDTS (2016-02-22 15:04:51.881612060) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.883221062) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.883548002) [info]: Failed to start EDTS server: 35 retries left
EDTS (2016-02-22 15:04:51.883873234) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.885343503) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.885669982) [info]: Failed to start EDTS server: 34 retries left
EDTS (2016-02-22 15:04:51.885952235) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.887234709) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.887719201) [info]: Failed to start EDTS server: 33 retries left
EDTS (2016-02-22 15:04:51.888192064) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.890778703) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.891135844) [info]: Failed to start EDTS server: 32 retries left
EDTS (2016-02-22 15:04:51.891467168) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.892987651) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.893333947) [info]: Failed to start EDTS server: 31 retries left
EDTS (2016-02-22 15:04:51.895379377) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.897141741) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.897463528) [info]: Failed to start EDTS server: 30 retries left
EDTS (2016-02-22 15:04:51.897781618) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.899441070) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.899794719) [info]: Failed to start EDTS server: 29 retries left
EDTS (2016-02-22 15:04:51.900133116) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.901654071) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.901981374) [info]: Failed to start EDTS server: 28 retries left
EDTS (2016-02-22 15:04:51.902289579) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.903782029) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.904326583) [info]: Failed to start EDTS server: 27 retries left
EDTS (2016-02-22 15:04:51.904730982) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.907388049) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.907722643) [info]: Failed to start EDTS server: 26 retries left
EDTS (2016-02-22 15:04:51.908037807) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.909562324) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.909843086) [info]: Failed to start EDTS server: 25 retries left
EDTS (2016-02-22 15:04:51.910161148) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.913550691) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.913882023) [info]: Failed to start EDTS server: 24 retries left
EDTS (2016-02-22 15:04:51.914198427) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.915694082) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.916033515) [info]: Failed to start EDTS server: 23 retries left
EDTS (2016-02-22 15:04:51.916353490) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.917893856) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.918249682) [info]: Failed to start EDTS server: 22 retries left
EDTS (2016-02-22 15:04:51.918590894) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.920062791) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.920302323) [info]: Failed to start EDTS server: 21 retries left
EDTS (2016-02-22 15:04:51.920514135) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.923460057) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.923880261) [info]: Failed to start EDTS server: 20 retries left
EDTS (2016-02-22 15:04:51.924212862) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.925739864) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.926082890) [info]: Failed to start EDTS server: 19 retries left
EDTS (2016-02-22 15:04:51.926426033) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.929851977) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.930202730) [info]: Failed to start EDTS server: 18 retries left
EDTS (2016-02-22 15:04:51.930516399) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.932173695) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.932505790) [info]: Failed to start EDTS server: 17 retries left
EDTS (2016-02-22 15:04:51.932833843) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.934340784) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.934656965) [info]: Failed to start EDTS server: 16 retries left
EDTS (2016-02-22 15:04:51.934973882) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.936434967) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.936739031) [info]: Failed to start EDTS server: 15 retries left
EDTS (2016-02-22 15:04:51.936985666) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.940193018) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.940490210) [info]: Failed to start EDTS server: 14 retries left
EDTS (2016-02-22 15:04:51.940823924) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.942481527) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.942830861) [info]: Failed to start EDTS server: 13 retries left
EDTS (2016-02-22 15:04:51.943136814) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.945931779) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.946163234) [info]: Failed to start EDTS server: 12 retries left
EDTS (2016-02-22 15:04:51.946332607) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.947200063) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.947402390) [info]: Failed to start EDTS server: 11 retries left
EDTS (2016-02-22 15:04:51.947584744) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.948821748) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.949099385) [info]: Failed to start EDTS server: 10 retries left
EDTS (2016-02-22 15:04:51.949273724) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.950283408) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.950491834) [info]: Failed to start EDTS server: 9 retries left
EDTS (2016-02-22 15:04:51.950645013) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.951643966) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.951908102) [info]: Failed to start EDTS server: 8 retries left
EDTS (2016-02-22 15:04:51.952232054) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.953559113) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.954049540) [info]: Failed to start EDTS server: 7 retries left
EDTS (2016-02-22 15:04:51.954504530) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.957101166) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.957429428) [info]: Failed to start EDTS server: 6 retries left
EDTS (2016-02-22 15:04:51.957738835) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.959324560) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.959658814) [info]: Failed to start EDTS server: 5 retries left
EDTS (2016-02-22 15:04:51.959989577) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.963223051) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.963564155) [info]: Failed to start EDTS server: 4 retries left
EDTS (2016-02-22 15:04:51.963895171) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.965576595) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.965917384) [info]: Failed to start EDTS server: 3 retries left
EDTS (2016-02-22 15:04:51.966253634) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.967877308) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.968218660) [info]: Failed to start EDTS server: 2 retries left
EDTS (2016-02-22 15:04:51.968557012) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.970030890) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.970366404) [info]: Failed to start EDTS server: 1 retries left
EDTS (2016-02-22 15:04:51.970585880) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:04:51.973567219) [debug]: Reply nil received for request to http://0:4587/nodes
File mode specification error: (error "EDTS: Could not start main server")
```

sleep-for

```
EDTS (2016-02-22 15:02:19.232205249) [info]: Initializing plugin edts_dialyzer
EDTS (2016-02-22 15:02:19.233218210) [info]: Initializing plugin edts_xref
For information about GNU Emacs and the GNU system, type C-h C-a.
EDTS (2016-02-22 15:02:19.493896913) [debug]: Setting up edts-mode in buffer sgw_gtpu_term_tests.erl
EDTS (2016-02-22 15:02:19.506374302) [debug]: Sending GET-request to http://0:4587/nodes
Using a proxy for http...
Package iswitchb is obsolete!
Reading machine info...done
EDTS (2016-02-22 15:02:19.572829772) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:02:19.573123262) [info]: Failed to start EDTS server: 150 retries left
EDTS (2016-02-22 15:02:19.573398454) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:02:19.577578448) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:02:19.578055021) [info]: Failed to start EDTS server: 149 retries left
EDTS (2016-02-22 15:02:19.578464728) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:02:19.579734626) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:02:19.651723586) [info]: Failed to start EDTS server: 148 retries left
EDTS (2016-02-22 15:02:19.651996222) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:02:19.654593961) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:02:19.793940019) [info]: Failed to start EDTS server: 147 retries left
EDTS (2016-02-22 15:02:19.794289917) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:02:19.798889922) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:02:19.885440120) [info]: Failed to start EDTS server: 146 retries left
EDTS (2016-02-22 15:02:19.885837063) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:02:19.891700541) [debug]: Reply nil received for request to http://0:4587/nodes
EDTS (2016-02-22 15:02:20.069495890) [info]: Failed to start EDTS server: 145 retries left
EDTS (2016-02-22 15:02:20.069772089) [debug]: Sending GET-request to http://0:4587/nodes
EDTS (2016-02-22 15:02:20.093379980) [debug]: Reply (200 OK) received for request to http://0:4587/nodes
EDTS (2016-02-22 15:02:20.095007162) [info]: Started EDTS server
```

henrikcaesar committed:
Workaround for sit-for bug during edts-api-start-server

There is a bug in sit-for that make the call return immediately
instead of waiting the wanted numer of seconds. This bug has been
around since emacs-23.2 and was fixed Wed, 29 Jul 2015, see:

  https://debbugs.gnu.org/cgi/bugreport.cgi?bug=15990
  https://debbugs.gnu.org/cgi/bugreport.cgi?bug=20935

This bug make EDTS fail to start with the message:

  EDTS: Could not start main server

since the retry loop in edts-api-start-server is executed too fast,
before the web server component has started.

Since a many emacs version are affected many users might get into
trouble for a long period of time. The fix not even been included in a
official emacs release yet. To work around the problem instead of
relying on sit-for also check how long time have passed in the while
loop condition.

henrikcaesar commented:
What do you think about merging this PR?

henrikcaesar commented:
Doesn't seem to be any interest in merging this PL and I have left the job where this was an issue.
